### ISSUE TITLE
Xournal++ crashes when using PipeWire

### ISSUE THREAD

**Issue by cdknight** [xournalpp/xournalpp#2916]
**Affects versions :**
 - OS: Linux (using Arch)
 - Desktop environment: KDE/KWin (X11)
 - Which version of libgtk do you use
     - 3.24.25
 - Version of Xournal++: [e.g. 1.0.16, specific commit, specific branch]
     - 1.0.20-hotfix
 - Installation method: Arch repositories

**Describe the bug**

If I install PipeWire (see https://wiki.archlinux.org/index.php/PipeWire for what I did), and then install the `pipewire-jack-dropin` and try launching Xournal++ so that it uses the PipeWire JACK libraries, it does not launch. Furthermore, if I uninstall the `pipewire-jack-dropin` and then try `pw-jack xournalpp`, I will also have Xournal++ fail to launch.

Crash logs from running in the terminal:

```
ALSA lib pcm_dsnoop.c:638:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1075:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_oss.c:377:(_snd_pcm_oss_open) Unknown field port
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_usb_stream.c:486:(_snd_pcm_usb_stream_open) Invalid type for card
ALSA lib pcm_dmix.c:1075:(snd_pcm_dmix_open) unable to open slave
xournalpp: src/hostapi/jack/pa_jack.c:517: BuildDeviceList: Assertion `match_info.rm_eo - match_info.rm_so < jack_client_name_size()' failed.

** (com.github.xournalpp.xournalpp:213991): WARNING **: 18:05:52.848: [Crash Handler] Crashed with signal 6

** (com.github.xournalpp.xournalpp:213991): WARNING **: 18:05:52.848: [Crash Handler] Wrote crash log to: /home/user/.xournalpp/errorlogs/errorlog.20210307-180552.log
```

**To Reproduce**

With `pipewire-jack-dropin`:

1. Install PipeWire on Arch Linux and the `pipewire-jack-dropin`
2. Open Xournal++
3. It will not launch

Without `pipewire-jack-dropin`

1. Install PipeWire
2. Run `pw-jack xournalpp` in the terminal
3. It will crash

**Expected behavior**
Xournal++ should work with PipeWire.

**Additional context**

Maybe I did PipeWire wrong or the `pipewire-jack-dropin` is broken, but I suspect it is not because `pw-jack xournalpp` doesn't work either. 

Also, perhaps this is a PipeWire bug and not a Xournal++ bug? AFAIK, PipeWire is supposed to provide API compatibility with JACK and PulseAudio, so if it's not working on some application, it would mean that PipeWire might be at fault?

**Comment by rolandlo** [comment]
I don't know who's at fault here, but it may also be a problem with `Portaudio`. The file `pa_jack.c` comes from `Portaudio`.

**Comment by Be-ing** [comment]
Hi, I saw this issue linked from my PortAudio pull request. You've ran into a [different bug in PipeWire](https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/788) which has already been fixed in the PipeWire master branch. My PortAudio PR fixes a bug where PortAudio reports devices that have regex tokens in their names as having 0 channels.

**Comment by rolandlo** [comment]
@Be-ing Thanks for informing us!

**Comment by Be-ing** [comment]
Nevertheless I'd appreciate if you could add your voice on that PortAudio PR urging the maintainers to merge it.

**Comment by Technius** [comment]
Closing since this is due to a bug in a dependency.

**Comment by cdknight** [comment]
I _think_ this is solved as of Pipewire 0.3.24.
